### PR TITLE
Lexer helpers

### PR DESCRIPTION
**Pull request by badochov**
Created general utilies.
Created helper type functions and variables for lexer.

### CODE REVIEW

File: src/lexer/re.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	"sort"
+)
+
+var newlineRe = regexp.MustCompile(`(\r\n|\r|\n)`)
+var whitespaceRe = regexp.MustCompile(`\s+`)
+var stringRe = regexp.MustCompile(`(?s)('([^'\\]*(?:\\.[^'\\]*)*)'|"([^"\\]*(?:\\.[^"\\]*)*)")`)
+var integerRe = regexp.MustCompile(`(?i)(0b(_?[0-1])+|0o(_?[0-7])+|0x(_?[\da-f])+|[1-9](_?\d)*|0(_?0)*)`)
+var floatRe = regexp.MustCompile(`(?i)(?<!\.)(\d+_)*\d+((\.(\d+_)*\d+)?e[+\-]?(\d+_)*\d+|\.(\d+_)*\d+)`)
+
+func countNewlines(value string) int {
```
Comment (jmolinski) [inline src/lexer/re.go:29]: suboptimal (regexes are slow), maybe add a TODO to rewrite this as a for loop in the future

**Comment by badochov** [comment]
golangci-lint fails but it is expected to happen and do not fix

File: src/lexer/tokens.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+	"fmt"
+	"github.com/gojinja/gojinja/src/utils/maps"
+	"github.com/gojinja/gojinja/src/utils/set"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const TOKEN_ADD = "add"
```
Comment (jmolinski) [inline src/lexer/tokens.go]: Mixed caps

https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#mixed-caps

File: src/lexer/tokens.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	if len(rev) != len(ops) {
+		_, _ = fmt.Fprintln(os.Stderr, "operators dropped")
+		os.Exit(1)
+	}
+	return rev
+}
+
+var operatorRe = getOperatorRe(operators)
+
+func getOperatorRe(ops map[string]string) *regexp.Regexp {
```
Comment (jmolinski) [inline src/lexer/tokens.go:108]: is it ever used in runtime? if so, MustCompile will be a problem

Comment (badochov) [reply src/lexer/tokens.go:108]: no it's only used here

**Review by jmolinski — APPROVED** [review]
LGTM

**Pull request merged**